fast_pauli/cpp/include/__summed_pauli_op.hpp

```diff
@@ -389,25 +563,40 @@ template <std::floating_point T> struct SummedPauliOp
         std::vector<std::complex<T>> expectation_vals_raw(n_pauli_strings() * n_data);
         Tensor<2> expectation_vals(expectation_vals_raw.data(), n_pauli_strings(), n_data);
 
-        if constexpr (std::is_same_v<ExecutionPolicy, std::execution::parallel_policy>)
+        if constexpr (is_parallel_execution_policy_v<ExecutionPolicy>)
         {
 
-#pragma omp parallel for schedule(static)
-            for (size_t j = 0; j < n_pauli_strings(); ++j)
+#pragma omp parallel
             {
-                std::mdspan expectation_vals_j = std::submdspan(expectation_vals, j, std::full_extent);
-                pauli_strings[j].expectation_value(expectation_vals_j, states);
-            }
+#pragma omp for schedule(static)
+                for (size_t j = 0; j < n_pauli_strings(); ++j)
+                {
+                    std::mdspan expectation_vals_j = std::submdspan(expectation_vals, j, std::full_extent);
+                    pauli_strings[j].expectation_value(expectation_vals_j, states);
```

Comment:
can also feed in `summed_coeffs(j)` as a third argument to expectation_value call. but that would require a contraction loop across coeffs in order to populate new vector

Reply:
So I don't think we should save `summed_coeffs` for the reasons I listed above, but if we did, it wouldn't work here because we don't want to sum over the `k` indices (since we want to output expectation values for each operator).